### DISCUSSION TITLE
Unable to discover any devices, even with manual

### DISCUSSION THREAD

**Discussion by schlosrat** [plasticrake/homebridge-tplink-smarthome#295] [General]
I've got 6 HS200 switches installed and working. By working, I mean only that they work in the Kasa app and on the wall if you manually operate them. They all show up in my router, naturally, so I've got their IP and MAC addresses. All of the HS200s are on 192.168.1.x. They're also all on the same guest WiFi network where I have my Nest and Rachio sprinkler (both of which discover and work just fine with their respective plugins).

I'm running Homebridge v1.6.0 on a Synology NAS with Node.js v16.16.0, homebridge-tplink-smarthome v7.0.1, and API v2.7. The Synology NAS is wired into the router, so it sees the Nest and Rachio on my guest WiFI and should see my HS200's too I think.

As the title says, I've been unable to get Homebridge to discover any of my HS200s. I've tried running this plugin as a child bridge and not as a child. I've tried various things in the broadcast setting (255.255.255.255 and 192.168.0.255 specifically), and I've tried setting Transport to TCP, UDP, and None. Currently, my settings look like this:

```
{
    "name": "TplinkSmarthome",
    "addCustomCharacteristics": false,
    "inUseThreshold": 0,
    "switchModels": [
        "HS200",
        "HS210"
    ],
    "discoveryPort": 0,
    "broadcast": "192.168.0.255",
    "pollingInterval": 10,
    "deviceTypes": [
        "plug",
        "bulb"
    ],
    "devices": [
        {
            "host": "192.168.1.8"
        },
        {
            "host": "192.168.1.9"
        },
        {
            "host": "192.168.1.10"
        },
        {
            "host": "192.168.1.11"
        },
        {
            "host": "192.168.1.12"
        },
        {
            "host": "192.168.1.13"
        }
    ],
    "timeout": 15,
    "waitTimeUpdate": 100,
    "platform": "TplinkSmarthome"
}

```
I've tried rebooting the switches, the NAS, and the Router. I've updated the switches to the latest FW, updated the router to the latest FW, and the NAS to the latest FW. I'm at a loss for what to try next. Any clues?

**Comment by schlosrat** [comment]
Update: Not sure what's different, but HB has managed to discover 2 of my switches. Does this normally take many hours?

I've installed tplink-smarthome-api and performed a search (this was after HB managed to detect 2 switches). Running this didn't change the detected devices.

```
tplink-smarthome-api search
Searching...
startDiscovery({
  discoveryInterval: 2000,
  discoveryTimeout: 10000,
  breakoutChildren: true,
  broadcast: '255.255.255.255'
})
```

The plugin config I've got now is this:

```
{
    "name": "TplinkSmarthome",
    "addCustomCharacteristics": false,
    "deviceTypes": [
        "plug",
        "bulb"
    ],
    "devices": [
        {
            "host": "192.168.1.8"
        },
        {
            "host": "192.168.1.9"
        },
        {
            "host": "192.168.1.10"
        },
        {
            "host": "192.168.1.11"
        },
        {
            "host": "192.168.1.12"
        },
        {
            "host": "192.168.1.13"
        }
    ],
    "transport": "udp",
    "_bridge": {
        "username": "0E:C8:05:CD:23:C2",
        "port": 40989
    },
    "platform": "TplinkSmarthome"
}
```

**Reply by schlosrat** [reply]
I've read online that with some Kasa HS200 HW/FW combinations the switch will not respond on port 9999, but will respond on port 80. In my case, my HS200s are all HW v 5.0, FW v 1.0.9, except for one that's HW v 5.0/ FW v 1.0.6. The two that are connecting are both FW v 1.0.9, but none will respond to ping and all are on static IP addresses visible in the router.

I noticed that in the manual discovery UI the default port is 9999, but it says you can override that. Did you know that's not quite true? The default part may be true, but I double-dog-dare you to try overriding it and see what happens. Perhaps there are some specific constraints for what a valid port is, but every value I've tried (9, 80, 999 - just poking around here) results in this error message in the HB Status window.

```
ConfigParseError: Error parsing config
    at parseConfig (/volume1/homebridge/node_modules/homebridge-tplink-smarthome/src/config.ts:262:11)
    at new TplinkSmarthomePlatform (/volume1/homebridge/node_modules/homebridge-tplink-smarthome/src/platform.ts:79:30)
    at ChildBridgeFork.startBridge (/volume1/homebridge/node_modules/homebridge/src/childBridgeFork.ts:150:42)
```

Does anyone here know what config.ts is looking for in the port field? I've looked at the config.json and the UI isn't writing something with syntax errors, so that's not the issue.

**Reply by schlosrat** [reply]
Well, I found the issue. It turned out to be a setting in my router that was preventing this plugin from reliably seeing my Kasa TP-Link devices. They all work perfectly now!

My router is a Netgear Nighthawk RAX70 running FW V1.0.13.128. This issue might also affect other Netgear routers or other brands, so I wanted to share it here in case anyone else runs into something similar.

In my wifi configuration, I have a guest 2.4GHz network with its own SSID that I use for all my IOT devices. When logging into my router in the Basic tab under the Guest Network there is a setting for "Allow guests to see each other and access my local network". I had previously configured that to be checked OFF in the thought that it would enable better security, but since I only connect IOT devices to this SSID I switched it to ON, and voila! All my Kasa TP-Link switches connect to the plugin now! This setting was also interfering with other natively HomeKit-compatible devices, so it's not just important to the operation of this plugin, but also the Apple Home app and other vendor apps, like my Hunter Simpleconnect smart fans.

![Router Config](https://user-images.githubusercontent.com/50781429/211068046-2820f8a8-5646-4788-a5e4-e5a1196aa01c.png)

**Reply by Frulexi** [reply]
Thank you so much i had this same exact problem, bless you sir

**Comment by jeffrey424** [comment]
Try setting the broadcast discover to 192.168.1.255. 

From what you have described about your network setup you have the devices on the 192.168.1.x subnet. 

I had an issue with discovery so I added my subnet which is 192.168.86.x. Once I added the broadcast setting for my specific subnet all of my devices showed up.

**Comment by schlosrat** [comment]
Thanks for the tip. I did try that already actually, but unfortunately, it didn't work in my case. Once I made the change to my router settings noted above things worked perfectly though, and it also made the detection of HomeKit-compatible devices possible. I ran into this fix when trying to sort out why I was unable to get some HomeKit-compatible fans and switches working. Fixing my router setting for those things made everything work for me, including the Kasa TP-Link switches I'd been having detection issues with.

My current settings for the broadcast discover is the default 255.255.255.255.

**Reply by jeffrey424** [reply]
Awesome and thanks for sharing the information about your router. 

Glad all of your devices are working now. 

I really like the TP-Link switches and outlets. The work great and do not break the bank.
